packages/devtools_app/lib/src/shared/config_specific/import_export/import_export.dart

```diff
@@ -79,17 +78,24 @@ class ImportController {
     }
 
     final connectedApp =
-        (devToolsSnapshot[DevToolsExportKeys.connectedApp.name] ??
-                <String, Object>{})
-            .cast<String, Object>();
+        OfflineConnectedApp.parse(devToolsSnapshot.connectedApp);
     offlineController
-      ..enterOfflineMode(offlineApp: OfflineConnectedApp.parse(connectedApp))
-      ..offlineDataJson = devToolsSnapshot;
+      ..enterOfflineMode(offlineApp: connectedApp)
+      ..offlineDataJson = devToolsSnapshot.json;
     notificationService.push(attemptingToImportMessage(activeScreenId));
     _pushSnapshotScreenForImport(activeScreenId);
   }
 }
 
+extension type _DevToolsSnapshot(Map<String, dynamic> json) {
+  Map<String, Object> get connectedApp {
+    final connectedApp = json[DevToolsExportKeys.connectedApp.name] as Map?;
+    return connectedApp == null ? {} : connectedApp.cast<String, Object>();
```

Comment:
Should these maps have a nullable Object?

Reply:
Done; thanks for catching